### DISCUSSION TITLE
Raspberry Pi Pico

### DISCUSSION THREAD

**Discussion by Jacques001** [revoxhere/duino-coin#520] [Q&A]
Hello. 

Does anybody know of a why to use a Raspberry Pi Pico to mine Duino coin?

**Comment by revoxhere** [comment]
There is no code for the Raspberry Pi pico yet as we don't have one

**Reply by Mod8** [reply]
When will we get one? I love this coin!

**Reply by ernesto75** [reply]
the pico is now supported, arduino code works on pico but will use only one core.

**Reply by dapiced** [reply]
Hi, i have flashed my **Raspberry pi** **pico** with file : **Arduino_Code.ino**.  Everything went ok but when i start the **AVR_Miner.py** it stuck at :

`sys0 AVR mining thread is starting using DUC0-S1A algorithm (/dev/ttyACM0)`

**DEBUG MODE** 

`18:51:50  sys0  AVR mining thread is starting using DUCO-S1A algorithm (/dev/ttyACM0)   `
`18:51:50.730439 DEBUG: /dev/ttyACM0: requested job from the server  `
`18:51:50.827409 DEBUG: Received: ['75064b256f9eed1bf9ef3850656d4cfbdec22715', '91c14375eda1a999c7e2c4f6e63a1bfea55b4e03', '6']`
`18:51:50.827667 DEBUG: 0Correct job received  `
`18:51:50.827741 DEBUG: /dev/ttyACM0: sending job to AVR  `


Something i forgot ?

Thanks
Dominic

**Comment by ernesto75** [comment]
So, we can now use Raspberry Pico to mine duinocoin, but only one core will be used. Can we use this discussion as a WIP for writing a custom miner for pico? I guess we can use micropython for this. Anyone willing to help? I know Python basics, but i need someone who knows C to translate the arduino code in micropython. There are libraries like hashlib and get_uniqueid_board_id() from pico sdk that we can use, and the dualcore can be used with threads. Also some mods on AVR_Miner.py would be needed to open multiple socket. I will write a draft in the next days, let me know if anyone want to jump in. @Jacques001

**Reply by dapiced** [reply]
Hi, pico is running fine. In Arduino IDE board manager, make sure you're using Mbed OS RP2040 version 2.3.1 and not 2.41...

—

**Reply by kosa6661987** [reply]
He didnt mind to detect or upload to board, just Run AVR miner, miner send job to Pico but didnt get respond.

**Reply by Jagadishdas** [reply]
I have limited knowledge about Pico

My understanding is that when Pico has nothing flashed to it - it will boot to USB mode.
When I placed microPython firmware on it - it shows as COM
Arduino IDE will not upload anything if it does not see COM device.

When I compiled the Duino code on 2.6.1 it failed the upload - no mater what I tried. The same with 2.3.1

"An error occurred while uploading the sketch"

Any idea what is wrong. I would appreciate help. Please help.

**Reply by ernesto75** [reply]
Try to press and hold the Bootsel button on the pico while you connect it to usb. The upload should work if you use Arduino Mbed OS RP2040 Board Version 2.3.1
